### PR TITLE
Add Hash MAC address get cli

### LINKED ISSUE #609: Certification: Add HashMAC address get cli
Test Harness requires a cli to get HashMac address which is used as extended mac address during commissioning process.

We can't use `extaddr` instead since Harness gets HashMac address before `joiner start`, so we need to add a new cli `hashmacaddr`.

### PR DESCRIPTION
**Pull request by chshu**
- move Hash Mac address calculation from joiner.cpp to mac.cpp

  - add hashmacaddr cli

resolves #609

### CODE REVIEW

File: src/core/openthread.cpp [hunk 1]
```diff
@@ -181,6 +181,11 @@ void otGetFactoryAssignedIeeeEui64(otInstance *aInstance, otExtAddress *aEui64)
     otPlatRadioGetIeeeEui64(aInstance, aEui64->m8);
 }
 
+void otGetHashMacAddress(otInstance *, otExtAddress *aHashMacAddress)
+{
+    sThreadNetif->GetMac().GetHashMacAddress(static_cast<Mac::ExtAddress *>(aHashMacAddress));
```
Comment (xiaom-GitHub) [inline src/core/openthread.cpp]: a typo? should be reinterpret_cast?

Comment (chshu) [inline src/core/openthread.cpp]: I think it's ok to use `static_cast` here, please let me know if you have any concern？Thanks.

Comment (xiaom-GitHub) [inline src/core/openthread.cpp]: ah, that's fine, it's an implicit conversion between base class and inherited class. You are correct. Please ignore this.

Comment (jwhui) [inline src/core/openthread.cpp]: Agree, using `static_cast` is preferred over `reinterpret_cast`.

**Pull request merged**